superbit_lensing/medsmaker/superbit/medsmaker_mocks.py

```diff
@@ -411,97 +454,208 @@ def make_psf_models(self, select_stars=False, use_coadd=False):
+                    truthfile=truthfilen,star_params=star_params)
+
+        else:
+            psfcat_name = self._select_stars_for_psf(sscat=imcat_ldac_name,\
+                    truthfile=None,star_params=star_params)
+
+        # Now run PIFF on that image and accompanying catalog
+        piff_config_arg = ''.join([sextractor_config_path,'piff.config'])
+        im_name = imagefile.replace('.fits','.sub.fits')
+        image_arg = ''.join(['input.image_file_name=',im_name])
+        psfcat_arg = ''.join(['input.cat_file_name=',psfcat_name])
+        output_name = imagefile.split('/')[-1].replace('.fits','.piff')
+        output_dir = os.path.join(self.data_dir,'piff-output')
+        full_output_name=os.path.join(output_dir,output_name)
+        output_arg = ''.join(['output.file_name=',output_name,' output.dir=',output_dir])
+        cmd = ' '.join(['piffify', piff_config_arg,image_arg,psfcat_arg, output_arg])
+
+        self.logprint("piff cmd is " + cmd)
+        os.system(cmd)
+
+        piff_extended = self.piff_extender(full_output_name)
+
+        return piff_extended
+
+    def piff_extender(self,piff_file,stamp_size=21):
+        """
+        Utility function to add the get_rec function expected
+        by the MEDS package
+
+        """
+        psf = piff.read(piff_file)
+
+        type_name = type(psf)
+
+        class PiffExtender(type_name):
+            '''
+            A helper class that adds functions expected by MEDS
+            '''
+
+            def __init__(self, type_name=None):
+
+                self.psf = None
+                self.single_psf = type_name
+
+                return
+
+            def get_rec(self,row,col):
+
+                #print('Working!!!')
+                fake_pex = self.psf.draw(x=col, y=row, stamp_size=stamp_size).array
+
+                return fake_pex
+
+            def get_center(self,row,col):
+
+                psf_shape = self.psf.draw(x=col,y=row,stamp_size=stamp_size).array.shape
+                cenpix_row = (psf_shape[0]-1)/2
+                cenpix_col = (psf_shape[1]-1)/2
+                cen = np.array([cenpix_row,cenpix_col])
+
+                return cen
+
+        psf_extended = PiffExtender(type_name)
+        psf_extended.psf = psf
+
+        return psf_extended
+
+    def _select_stars_for_psf(self,sscat,truthfile=None,starkeys=None,star_params=None):
         '''
-        Method to obtain stars from SExtractor catalog using the truth catalog from GalSim 
+        Method to obtain stars from SExtractor catalog using the truth catalog from GalSim
             sscat : input ldac-format catalog from which to select stars
             truthcat : the simulation truth catalog written out by GalSim
         '''
 
-        # Read in truthfile, obtain stars with redshift cut
-        truthcat = Table.read(truthfile,format='ascii')
-        stars=truthcat[truthcat['redshift']==0] 
+        
+        try:
```

Comment:
This is a good solution to what I brought up above - use a try/catch for the annoying SE hdu=1/2 issue.